### DISCUSSION TITLE
redirect to custom domain fails

### DISCUSSION THREAD

**Discussion by mluerig** [honkit/honkit#292] [Q&A]
I am hosting a honkit project with netlify under a custom domain (www.wiki.asellus.org). I have a added a netlify rewrite-rule to have the site appear under www.asellus.org/wiki using a `netlify.toml` file with this content:

    [[redirects]]
      from = "/wiki/*"
      to = "https://wiki.asellus.org/:splat"
      status = 200
      force = true

 The subdomain is working, but the rewrite is not working as expected: the css is missing, and all links point to the base url without "wiki" in between. In jekyll, this is solved by specifying the `baseurl` option in the config - is there a similar option for honkit?

**Comment by azu** [comment]
`https://www.asellus.org/wiki/` seems work.
last slash `/` is related to this issue.

When access `https://www.asellus.org/wiki`, Load `https://www.asellus.org/gitbook/style.css`. 
It is 404.

When access `https://www.asellus.org/wiki/`, Load `https://www.asellus.org/wiki/gitbook/style.css`
It is ok.

**Comment by mluerig** [comment]
thanks - I hadn't noticed. I now tried with additional netlify rewrite rules, but it may not work because domain level rewrites from asellus.org/wiki -> asellus.org/wiki/ would engage in an infinite loop with the redirect above. 

however, the netlify support found the [basepath](https://github.com/honkit/honkit/search?q=basepath) - can this be used for my purpose? asked differently, is there a way to solve the slash issue from the honkit side?

**Comment by mluerig** [comment]
is there no such option for honkit? would really help tremendously...

**Comment by azu** [comment]
Unfortunately, `baseurl` option is not implemented.
HonKit(GitBook v1) always handle the path as relative path. 
https://github.com/honkit/honkit/blob/4a439e5aff23b0722b93ab30186f2a30872a12e4/packages/honkit/src/output/website/onPage.ts#L37 

> is there a way to solve the slash issue from the honkit side?

It is difficult that honkit reconize it, I think.

Another option.

- handle the path as absolute path.
  - It requires `baseurl` option. 
  
create-react-app use similar approach. 
create-react-app handle the path as absolute path if `homepage` is defined.

- [Deployment | Create React App](https://create-react-app.dev/docs/deployment/#step-1-add-homepage-to-packagejson)
- [create-react-app/getPublicUrlOrPath.js at bb64e31a81eb12d688c14713dce812143688750a · facebook/create-react-app](https://github.com/facebook/create-react-app/blob/bb64e31a81eb12d688c14713dce812143688750a/packages/react-dev-utils/getPublicUrlOrPath.js#L25)

📝 `baseurl` option allow user to set different url with actual. In this case, it will be failed as [CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) error because HonKit use XMLHttpRequest.
